scripts/provision/docker.sh

```diff
@@ -49,11 +49,31 @@
 
 NAME=hyperledger/fabric-baseimage
 RELEASE=`uname -m`-$1
-FQN=$NAME:$RELEASE
+DOCKERHUB_NAME=$NAME:$RELEASE
 
 CURDIR=`dirname $0`
 
-docker pull $FQN
+echo "BUILD-CACHE: Pulling \"$DOCKERHUB_NAME\" from dockerhub.."
+docker pull $DOCKERHUB_NAME
+docker inspect $DOCKERHUB_NAME 2>&1 > /dev/null
+if [ "$?" == "0" ]; then
+    echo "BUILD-CACHE: Success!"
+    BASENAME=$DOCKERHUB_NAME
+else
+    echo "BUILD-CACHE: WARNING - Build-cache unavailable, attempting local build but this should be fixed ASAP"
+    (cd $CURDIR/../../baseimage && make docker DOCKER_TAG=localbuild)
```

Comment:
This probably needs to have its retval checked for error.  We wouldn't want to proceed if this fails.